docs/configuration/configuration.md

```diff
@@ -2145,3 +2187,28 @@ tls_config:
 There is a list of
 [integrations](https://prometheus.io/docs/operating/integrations/#remote-endpoints-and-storage)
 with this feature.
+
+### `oauth2`
+
+OAuth 2.0 authentication using the client credentials grant type.
+Fetches the token from the specified endpoint with the client access and secret keys.
```

Comment:
```suggestion
With this section, Prometheus fetches an access token from the specified endpoint with the given client access and secret keys.
```

What about something in these lines?

Reply:
I would remove `With this section`, but the rest sounds good.